scripts/backport.py

```diff
@@ -305,55 +352,71 @@ def __init__(self, pygithub_pr_, issue_number_):
-        print(
-            f'Will not backport the PR #{original_pr.number}: "{original_pr.title}" because it contains multiple commits.'
-        )
-        original_pr.add_to_labels("pr-auto-backport-failed")
-        continue
-
     # Try to cherry-pick the commits.
     git_check(
         f"checkout --quiet --detach {target_remote}/{backport_target} > /dev/null"
     )
 
+    commit_shas = [commit.sha for commit in pr_info.pygithub_commits]
     if git_returncode(f"cherry-pick --quiet -m 1 -x {' '.join(commit_shas)}") != 0:
+        details = f"### Git status\n\n```\n{git_output('status')}\n```"
         git_check("cherry-pick --abort")
-        print(
-            f'Conflict while backporting pull request #{original_pr.number}: "{original_pr.title}"'
-        )
-        original_pr.add_to_labels("pr-auto-backport-failed")
+        report_backport_not_done(original_pr, "cherry-pick failed", details)
         continue
 
-    # Push the branch and create the backport PR
+    # Push the backport branch.
     git_check(
         f"push --quiet {target_remote} @:refs/heads/{backport_branch} > /dev/null 2>&1"
     )
 
-    original_description = original_pr.body
-    # Comment out the Github issue reference keywords like 'Fixes #1234', to
-    # avoid having multiple PRs saying they fix a given issue. The backport PR
-    # is going to reference the fixed issue as "Original issue #xxxx".
+    # Prepare description for the backport PR.
+    backport_description = (
+        f"This is an automated backport of #{original_pr.number}: {original_pr.title}."
+    )
+
+    if pr_info.issue_number:
+        backport_description += f"\nThe original issue is #{pr_info.issue_number}."
+
+    # Do not merge the PR automatically if it changes some particularly
+    # conflict-prone files that are better to review manually. Also mention this
+    # in the description.
+    changed_files = {file.filename for file in original_pr.get_files()}
+    stopper_files = changed_files.intersection(
+        ["sql/updates/latest-dev.sql", "sql/updates/reverse-dev.sql"]
+    )
+    if stopper_files:
+        backport_description += (
+            "\n"
+            f"This PR will not be merged automatically, because it modifies '{list(stopper_files)[0]}' "
+            "which is conflict-prone. Please review these changes manually."
+        )
+    else:
+        backport_description += (
+            "\n"
+            "This PR will be merged automatically after all the relevant CI checks pass."
+        )
+
+    backport_description += (
+        " If this fix should not be backported, or will be backported manually, "
+        "just close this PR. You can use the backport branch to add your "
+        "changes, it won't be modified automatically anymore."
+        "\n"
+        "\n"
+        "For more details, please see the [documentation]"
+        "(https://github.com/timescale/eng-database/wiki/Releasing-TimescaleDB#automated-cherry-picking-of-bug-fixes)"
```

Comment:
This is a private repository. Should we link to internal repositories? Looks good for now, but we might want to move this description to the public repository to make it available for everybody.